src/server/routes/units.js

```diff
@@ -231,12 +231,17 @@ router.post('/delete', adminAuthMiddleware('delete units'), async (req, res) =>
 	} else {
 		const conn = getConnection();
 		try {
-			// Don't worry about checking if the unit already exists
-			// Just try to delete it to save the extra database call, since the database will return an error anyway if the row does not exist
-			await Unit.delete(req.body.id, conn);
-			success(res, 'Successfully deleted conversion');
+			await conn.tx(async t => {
+				// Update meters that use this unit as default graphic unit
```

Comment:
Two thoughts. First, you reuse ``req.body.id`` several times. It might be better to create a const and use that. Second, you are directly calling the SQL in the route as was done in src/server/routes/conversions.js. While it was done there, it would be better to do it in the meter/group model. Also, it duplicates the code (why having in the model is better). Thus, I think a new function in both meter and group to update based on the provided default graphic unit should be created and used in both files.